src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOf.Fixer.cs

```diff
@@ -0,0 +1,126 @@
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public sealed class PreferStringContainsOverIndexOfFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(PreferStringContainsOverIndexOfAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Document doc = context.Document;
+            CancellationToken cancellationToken = context.CancellationToken;
+            SyntaxNode root = await doc.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            if (root.FindNode(context.Span) is SyntaxNode expression)
+            {
+                SemanticModel semanticModel = await doc.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var compilation = semanticModel.Compilation;
+                if (!compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemString, out INamedTypeSymbol? stringType) ||
+                    !compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemChar, out INamedTypeSymbol? charType) ||
+                    !compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemStringComparison, out INamedTypeSymbol? stringComparisonType))
+                {
+                    return;
+                }
+
+                var operation = semanticModel.GetOperation(expression, cancellationToken);
+                // Not offering a code-fix for the variable declaration case
+                if (!(operation is IBinaryOperation binaryOperation))
+                {
+                    return;
+                }
+
+                IInvocationOperation invocationOperation;
+                IOperation otherOperation;
+                if (binaryOperation.LeftOperand is IInvocationOperation invocationOperationOperand)
+                {
+                    invocationOperation = invocationOperationOperand;
+                    otherOperation = binaryOperation.RightOperand;
+                }
+                else
+                {
+                    invocationOperation = (IInvocationOperation)binaryOperation.RightOperand;
+                    otherOperation = binaryOperation.LeftOperand;
+                }
+
+                var instanceOperation = invocationOperation.Instance;
```

Comment:
Defensive check to bail out for all cases except argument length 1 and 2? These seem to be the only cases handled in `ReplaceBinaryOperationWithContains`.

```suggestion
                switch (invocationOperation.Arguments.Length)
                {
                    case 1:
                    case 2:
                        break;
                    default:
                        return;
                }

                var instanceOperation = invocationOperation.Instance;
```